### PR TITLE
[Package] Separate single-shot and pipeline API.

### PR DESCRIPTION
**Pull request by gichan-jang**
- `machine-learning-inference` is separated to `machine-learning-inference-single` and `machine-learning-inference-pipeline`.
 - Move `_ml_validate_model_file` to `ml-api-inference-single`

Signed-off-by: Gichan Jang <gichan2.jang@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
```
[   69s] /home/abuild/rpmbuild/BUILD/capi-machine-learning-inference-1.8.0/c/src/ml-api-inference-internal.c:16:10: fatal error: nnstreamer_plugin_api_util.h: No such file or directory
[   69s]    16 | #include <nnstreamer_plugin_api_util.h>
[   69s]       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[   69s] compilation terminated.
```

File: packaging/machine-learning-api.spec [hunk 1]
```diff
@@ -64,6 +64,8 @@ Source1001:	capi-machine-learning-inference.manifest
 
 ## Define build requirements ##
 Requires:	nnstreamer
+Requires:	capi-machine-learning-inference-sinlge = %{version}-%{release}
```
Comment (myungjoo) [inline packaging/machine-learning-api.spec:67]: TYPO: "sinlge"

**Review by myungjoo — COMMENTED** [review]
Test the final product with a simple test app using nnstreamer-single-api.rpm in Tizen device (or in gbs). It will fail.

**Review by myungjoo — CHANGES_REQUESTED** [review]
Backward compatibility is broken.

An app with ```Requires: capi-machine-learning-inference-devel``` will be broken.

Add dependency from devel to single-devel and pipeline-devel so that previous applications can be kept built after this commit.

**Comment by gichan-jang** [comment]
> Backward compatibility is broken.
> 
> An app with `Requires: capi-machine-learning-inference-devel` will be broken.
> 
> Add dependency from devel to single-devel and pipeline-devel so that previous applications can be kept built after this commit.


https://github.com/gichan-jang/api/blob/1899b0fb2651e11f86f1f2308c1d97e44e881174/packaging/machine-learning-api.spec#L143-L151
I think dependencies has already been made. Please let me know if I missed something.

**Review by myungjoo — APPROVED** [review]
LGTM

**Comment by myungjoo** [comment]
> > Backward compatibility is broken.
> > An app with `Requires: capi-machine-learning-inference-devel` will be broken.
> > Add dependency from devel to single-devel and pipeline-devel so that previous applications can be kept built after this commit.
> 
> https://github.com/gichan-jang/api/blob/1899b0fb2651e11f86f1f2308c1d97e44e881174/packaging/machine-learning-api.spec#L143-L151 I think dependencies has already been made. Please let me know if I missed something. I built this package using gbs, and built the `nnstreamer-example`, it was passed.

My bad. I missed "devel" package was "capi-machine-learning-inference-devel".

File: c/meson.build [hunk 2]
```diff
@@ -113,18 +164,34 @@ if get_option('default_library') == 'static'
...
 nns_capi_dep = declare_dependency(link_with: nns_capi_lib,
-  dependencies: nns_capi_deps,
+  dependencies: [nns_capi_deps,nns_capi_single_dep, nns_capi_pipeline_dep],
   include_directories: nns_capi_include,
 )
 
 ml_inf_conf = configuration_data()
 ml_inf_conf.merge_from(api_conf)
-ml_inf_conf.set('ML_INFERENCE_REQUIRE', 'capi-ml-common')
+ml_inf_conf.set('ML_INFERENCE_REQUIRE', 'capi-nnstreamer-single capi-nnstreamer-pipeline')
```
Comment (zhoonit) [inline c/meson.build:173]: ML_INFERENCE_REQUIRE requires those two

**Pull request merged**